aws-java-eks-minimal/src/main/java/com/pulumi/example/eksminimal/App.java

```diff
@@ -18,32 +19,61 @@ public static void main(String[] args) {
     }
 
     private static void stack(Context ctx) {
-        var vpcIdOutput = Ec2Functions.getVpc(
+
+        // Find the default VPC identifier.
+        final var defaultVpcId = Ec2Functions.getVpc(
                 GetVpcArgs.builder().default_(true).build()
         ).applyValue(GetVpcResult::id);
-        ctx.export("vpcIdOutput", vpcIdOutput);
-
-        var subnetIdsOutput = vpcIdOutput
-                .apply(vpcId -> Ec2Functions.getSubnetIds(GetSubnetIdsArgs.builder()
-                                .vpcId(vpcId)
-                                .build()))
-                .applyValue(getSubnetIdsResult ->
-                        getSubnetIdsResult.ids()
-                                .stream()
-                                .sorted()
-                                .limit(2)
-                                .collect(Collectors.toList()));
-
-        ctx.export("subnetIdsOutput", subnetIdsOutput.applyValue(vs -> String.join(",", vs)));
-
-        var cluster = new Cluster("my-cluster", ClusterArgs.builder()
-                .vpcId(vpcIdOutput)
-                .subnetIds(subnetIdsOutput)
+
+        // Export it for visibility via `pulumi stack output`.
+        ctx.export("defaultVpcId", defaultVpcId);
+
+        // Find all the subnets in the default VPC.
+        final var allSubnetIds = Ec2Functions.getSubnets(GetSubnetsArgs.builder()
+                        .filters(GetSubnetsFilterArgs.builder()
```

Comment:
getSubnetIds returned deprecation warnings. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets is recommended to be used instead. Its not obvious from the types but this is how it's to be used.